Composer/packages/client/src/pages/botProject/PublishTargets.tsx

```diff
@@ -156,37 +141,56 @@ export const PublishTargets: React.FC<PublishTargetsProps> = (props) => {
     [publishTargets, projectId]
   );
 
-  useEffect(() => {
+  // add profile
+  const openAddProfileDialog = useCallback(() => {
     setDialogProps({
-      title: formatMessage('Add a publish profile'),
+      ...PublishProfileDialog.ADD_PROFILE,
       type: DialogType.normal,
-      children: (
-        <CreatePublishTarget
-          closeDialog={() => setAddDialogHidden(true)}
-          current={null}
-          targets={publishTargets || []}
-          types={publishTypes}
-          updateSettings={savePublishTarget}
-        />
-      ),
     });
+    setDialogChildren(
```

Comment:
It's better to control the DialogChildren's visibility by defining a boolean state rather than saving JSX to state.